Frontend/frontend/src/Util/User.js

```diff
@@ -0,0 +1,23 @@
+export default class User {
+    
+    static setUserState(newState) {
+        this.userState = newState;
+
+        localStorage.setItem("flatmates_app_user", JSON.stringify(newState));
+    }
+
+    static clearUserState(newState) {
+        this.userState = undefined;
+
+        localStorage.removeItem("flatmates_app_user");
+    }
+
+    static getUserState() {
+        return this.userState;
+    }
+}
+
+try {
+    User.userState = localStorage.getItem("flatmates_app_user") ?
+        JSON.parse(localStorage.getItem("flatmates_app_user")) : undefined;
+} catch (e) {}
```

Comment:
Is there no desired behaviour if the exception is thrown?

Reply:
No, if then user object is not valid, then the desired behaviour is that userState is left as is (undefined).